plugins/services/src/js/events/ServiceDeleteActions.js

```diff
@@ -0,0 +1,106 @@
+import { RequestUtil } from "mesosphere-shared-reactjs";
+import AppDispatcher from "#SRC/js/events/AppDispatcher";
+import CosmosPackagesActions from "#SRC/js/events/CosmosPackagesActions";
+import {
+  REQUEST_COSMOS_PACKAGE_UNINSTALL_ERROR,
+  REQUEST_COSMOS_PACKAGE_UNINSTALL_SUCCESS
+} from "#SRC/js/constants/ActionTypes";
+
+import {
+  REQUEST_MARATHON_SERVICE_DELETE_ERROR
+} from "../constants/ActionTypes";
+import MarathonActions from "./MarathonActions";
+import Framework from "../structs/Framework";
+
+const onFrameworkDelete = function(action, subscriptionId, resolve, reject) {
+  if (action.type === REQUEST_COSMOS_PACKAGE_UNINSTALL_SUCCESS) {
+    AppDispatcher.unregister(subscriptionId);
+    resolve();
+
+    return;
+  }
+
+  if (action.type === REQUEST_COSMOS_PACKAGE_UNINSTALL_ERROR) {
+    AppDispatcher.unregister(subscriptionId);
+    reject(RequestUtil.getErrorFromXHR(action.xhr));
+
+    return;
+  }
+};
+
+function uninstallFramework(framework) {
+  return new Promise(function(resolve, reject) {
+    const subscriptionId = AppDispatcher.register(payload => {
+      const { action } = payload;
+      if (
+        action.packageName === framework.getPackageName() &&
+        action.appId === framework.getId()
+      ) {
+        onFrameworkDelete(action, subscriptionId, resolve, reject);
+      }
+    });
+    CosmosPackagesActions.uninstallPackage(
+      framework.getPackageName(),
+      framework.getId()
+    );
+  });
+}
+
+function deleteFramework(framework) {
+  uninstallFramework(framework)
+    .then(function() {
+      MarathonActions.deleteService(framework);
+    })
+    .catch(function(error) {
+      AppDispatcher.handleServerAction({
+        type: REQUEST_MARATHON_SERVICE_DELETE_ERROR,
+        data: error
+      });
+    });
+}
+
+const ServiceDeleteActions = {
+  // This part is responsible for uninstalling the frameworks before deleting
```

Comment:
Maybe add a note/link on why it's disabled for now. And the conditions in which it could be enabled? Or remove this commented out code all together?

Reply:
I removed this altogether